### PR TITLE
feat(sprints): created unit tests for the sprints' service and controller methods

### PR DESCRIPTION
**Pull request by timDeHof**
# Description

 
Added comprehensive unit tests for `sprints.controller` and `sprints.service` following the testing guidelines. Also created centralized mock data system for improved test maintainability. 

## Issue link

Fixes # [(86azu6kx1)](https://app.clickup.com/t/86azu6kx1)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- all unit tests pass
- verify that mock data consistency
- check if unit tests followed unit vs e2e test separation rules

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Comment by cherylli** [comment]
All tests passed but I'm getting this, normal seeding (`prisma db seed`) is also fine
image

**Comment by timDeHof** [comment]
> All tests passed but I'm getting this, normal seeding (`prisma db seed`) is also fine image

I think this comes from my test in `sprints.service`, where I test for database errors during response group creation since the `addCheckinFormResponse` method would console log the error. I have added a jest's `spyOn` function to catch the errors. The Spy might fail if we change to something other than "log" like "error".

File: src/sprints/sprints.controller.spec.ts [hunk 1]
```diff
@@ -1,25 +1,1261 @@
...
+    notes: "Please come prepared",
+  };
+
+  const mockUpdatedMeetingDto: UpdateTeamMeetingDto = {
+    title: "Updated Sprint Meeting",
+    description: "Updated Planning session for sprint 1" ?? null,
+    meetingLink: "http://meet.google.com/123Updated" ?? null,
+    dateTime: mockDate ?? null,
+    notes: "Updated Please come prepared" ?? null,
+  };
```
Comment (Ajen07) [inline src/sprints/sprints.controller.spec.ts:51]: I think we don't  need  to use  `??` operator  as the left hand side will never be null

**Review by Ajen07 — CHANGES_REQUESTED** [review]

**Review by Ajen07 — DISMISSED** [review]

**Comment by cherylli** [comment]
All tests passed and worked. I'm not requesting any changes but have a few questions/comments
- "Also created centralized mock data system for improved test maintainability." Does that mean mock-data.ts is a shared file? Should we update the other tests to use this?

**Comment by timDeHof** [comment]
> All tests passed and worked. I'm not requesting any changes but have a few questions/comments
> 
> * "Also created centralized mock data system for improved test maintainability." Does that mean mock-data.ts is a shared file? Should we update the other tests to use this?

Yes, My idea for `mock-data.ts` is to be shared file to generate test items.

**Comment by cherylli** [comment]
> > All tests passed and worked. I'm not requesting any changes but have a few questions/comments
> > 
> > * "Also created centralized mock data system for improved test maintainability." Does that mean mock-data.ts is a shared file? Should we update the other tests to use this?
> 
> Yes, My idea for `mock-data.ts` is to be shared file to generate test items.

In this case, should we move it under some sort of shared folder instead of under /sprint module?

I will also add some low priority tasks to update old tests to use this, more importantly all new tests should use the shared mock data

**Comment by timDeHof** [comment]
I'll move the file to the shared folder and let you know when its ready.

**Comment by timDeHof** [comment]
I have moved the `mock-data.ts` to `/global/mocks`

**Review by cherylli — APPROVED** [review]
thanks, re ran all the tests with new changes, all passed

File: src/sprints/sprints.service.spec.ts [hunk 2]
```diff
@@ -1,29 +1,2422 @@
...
+                    );
+
+                expect(result).toEqual(mockForm);
+                expect(prismaMock.teamMeeting.findUnique).toHaveBeenCalledWith({
+                    where: { id: meetingId },
+                    select: { voyageTeamId: true },
+                });
+
+                expect(prismaMock.form.findUnique).toHaveBeenCalledWith({
+                    where: { id: formId },
```
[the remark this answers is not in the document]
Comment (cherylli) [reply src/sprints/sprints.service.spec.ts:1329]: this was from last time he didn't get sent 😂

**Pull request merged**